### PR TITLE
[inference][trt] add elementwise input data type check

### PR DESCRIPTION
**Pull request by zhangjun**
### PR types

Others
### PR changes

Others
### Describe

- Check the data type of input tensor for Elementwise OP
     - ElementWiseOperation::kLESS/kGREATER do not support boolean datatype
     - ElementWiseOperation::kPOW do not support int32 datatype
     - ElementWiseOperation::kADD/kMUL/kSUB/kDIV/kPOW/kMIN/kMAX/kFLOOR_DIV do not support boolean datatype

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/ir/inference/test_trt_convert_elementwise.py [hunk 1]
```diff
@@ -896,5 +896,157 @@ def test(self):
         self.run_test()
 
 
+class TrtConvertElementwiseTest_two_input_skip_case(TrtLayerAutoScanTest):
```
Comment (xiaoxiaohehe001) [inline python/paddle/fluid/tests/unittests/ir/inference/test_trt_convert_elementwise.py]: 类名称建议都改成大驼峰命名 TrtConvertElementwiseTwoInputSkipCaseTest

File: paddle/fluid/inference/tensorrt/op_teller.cc [hunk 2]
```diff
@@ -1365,16 +1365,26 @@ struct SimpleOpTypeSetTeller : public Teller {
...
+          return false;
+        }
+      }
+      if (op_type == "less_than" || op_type == "greater_than" ||
+          op_type == "less_equal") {
+        if (x_dtype == framework::proto::VarType::BOOL ||
+            y_dtype == framework::proto::VarType::BOOL) {
+          VLOG(3)
+              << "ElementWiseOperation::kLESS/ElementWiseOperation::kGREATER "
+                 "do not support boolean datatype..";
```
Comment (xiaoxiaohehe001) [inline paddle/fluid/inference/tensorrt/op_teller.cc]: 多加个句号.

File: python/paddle/fluid/tests/unittests/ir/inference/test_trt_convert_equal.py [hunk 3]
```diff
@@ -23,7 +23,7 @@
 import paddle.inference as paddle_infer
 
 
-class TrtConvertElementwiseTest_one_input_corner_case(TrtLayerAutoScanTest):
+class TrtConvertEqual_one_input_corner_case(TrtLayerAutoScanTest):
```
Comment (xiaoxiaohehe001) [inline python/paddle/fluid/tests/unittests/ir/inference/test_trt_convert_equal.py]: 同上

**Review by xiaoxiaohehe001 — APPROVED** [review]
LGTM

**Review by jiweibo — APPROVED** [review]
LGTM

**Pull request merged**